components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -71,6 +74,9 @@ constexpr auto kVideoTrackHosts =
                                                  "www.ted.com",
                                              });
 
+constexpr char kSearchSummarizerKeyPattern[] =
+    "<meta name=\"summarizer-key\" content=\"(.*)\"></meta>";
```

Comment:
`.+` might be better instead of `.*` if you are looking for nonempty keys

Reply:
either way i think you need a `?` at the end of the captured group, otherwise HTML like this:

`<meta name="summarizer-key" content="hi"></meta><meta id=""></meta>`

would return this:

`hi"></meta><meta id="`

instead of `hi` which is presumably what you want

Reply:
`<meta name="summarizer-key" content="hi"></meta><meta id=""></meta>` would be two separate entries in `snapshot.tree_data.metadata`, so I think it's behaving correctly in this case.
Will change to `.+`.

Reply:
@yrliou so i guess that means each entry that this regex is matching against only contains a single <meta> tag? i think you could still have a problem, for instance:

`<meta name="summarizer-key" content="foo" other="bar"></meta>`

would capture `foo" other="bar` instead of just `foo`

i assume we "trust" that the search team will not other attributes, but would be nice to make this more robust against HTML attribute injection either accidental or malicious.

Reply:
i guess something like this might work

`<meta name=\"summarizer-key\" content=\"(.+?)\".*></meta>`

Reply:
@diracdeltas Makes sense, how about this:
`<meta name="summarizer-key" content="(.*?)".*?><\/meta>` and check if key is empty in `OnSearchSummarizerKeyFetched`?

Reply:
I didn't use (.+?) because in `<meta name="summarizer-key" content="" other="bar"></meta>` case, it would match to `" other=`

Reply:
sgtm

Reply:
@diracdeltas https://github.com/brave/brave-core/pull/25257/commits/d2aeb9973edcca939f23efcff2df0ca026a55895
I have to enforce matching {} from json object because there will be double quotes inside, without it it would try to only match until {, please check the regular expression in this commit, thanks.

Reply:
lgtm